pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -779,25 +774,33 @@ func configureDhcpConfigV6Addresses(b *MasqueradeBindMechanism, err error) error
 		b.vmIPv6NetworkCIDR = api.DefaultVMIpv6CIDR
 	}
 
-	defaultGatewayIpv6, vmIpv6, err := b.handler.GetHostAndGwAddressesFromCIDR(b.vmIPv6NetworkCIDR)
+	gatewayAddr, vmAddr, err := b.generateGatewayAndVmIPAddrs(b.vmIPv6NetworkCIDR)
 	if err != nil {
-		log.Log.Reason(err).Errorf("failed to get gw and vm available ipv6 addresses from CIDR %s", b.vmIPv6NetworkCIDR)
 		return err
 	}
 
-	gatewayIpv6Addr, err := b.handler.ParseAddr(defaultGatewayIpv6)
+	b.dhcpConfig.GatewayIpv6 = gatewayAddr.IP.To16()
+	b.dhcpConfig.IPv6 = *vmAddr
+	b.gatewayIpv6Addr = gatewayAddr
+	b.ipv6Addr = *vmAddr
+	return nil
+}
+
+func (b *MasqueradeBindMechanism) generateGatewayAndVmIPAddrs(cidr string) (*netlink.Addr, *netlink.Addr, error) {
```

Comment:
make sense to pass just the ip family as a parameter. `b` has all the other info, including the `cidr`.